src/NodaTime/HighPerformance/Instant64.cs

```diff
@@ -0,0 +1,561 @@
+        return days >= 0
+            ? new Instant64(days * NanosecondsPerDay + nanoOfDay)
+            : new Instant64((days + 1) * NanosecondsPerDay + nanoOfDay - NanosecondsPerDay);
+    }
+
+    /// <summary>
+    /// Returns the later instant of the given two.
+    /// </summary>
+    /// <param name="x">The first instant to compare.</param>
+    /// <param name="y">The second instant to compare.</param>
+    /// <returns>The later instant of <paramref name="x"/> or <paramref name="y"/>.</returns>
+    public static Instant64 Max(Instant64 x, Instant64 y) => x > y ? x : y;
+
+    /// <summary>
+    /// Returns the earlier instant of the given two.
+    /// </summary>
+    /// <param name="x">The first instant to compare.</param>
+    /// <param name="y">The second instant to compare.</param>
+    /// <returns>The earlier instant of <paramref name="x"/> or <paramref name="y"/>.</returns>
+    public static Instant64 Min(Instant64 x, Instant64 y) => x < y ? x : y;
+
+    /// <summary>
+    /// Returns a <see cref="System.String" /> that represents this instance.
+    /// </summary>
+    /// <returns>
+    /// The value of the current instance, converted to an <see cref="Instant"/>, in the default format pattern ("g"), using the current thread's
+    /// culture to obtain a format provider.
+    /// </returns>
+    public override string ToString() => ToInstant().ToString();
+
+    /// <summary>
+    /// Formats the value of the current instance, converted to an <see cref="Instant"/>, using the specified pattern.
+    /// </summary>
+    /// <returns>
+    /// A <see cref="System.String" /> containing the value of the current instance in the specified format.
+    /// </returns>
+    /// <param name="patternText">The <see cref="System.String" /> specifying the pattern to use,
+    /// or null to use the default format pattern ("g").
+    /// </param>
+    /// <param name="formatProvider">The <see cref="System.IFormatProvider" /> to use when formatting the value,
+    /// or null to use the current thread's culture to obtain a format provider.
+    /// </param>
+    /// <filterpriority>2</filterpriority>
+    public string ToString(string? patternText, IFormatProvider? formatProvider) =>
+        ToInstant().ToString(patternText, formatProvider);
+
+    /// <summary>
+    /// Indicates whether the value of this instant is equal to the value of the specified instant.
+    /// See the type documentation for a description of equality semantics.
+    /// </summary>
+    /// <param name="other">The value to compare with this instance.</param>
+    /// <returns>
+    /// true if the value of this instant is equal to the value of the <paramref name="other" /> parameter;
+    /// otherwise, false.
+    /// </returns>
+    public bool Equals(Instant64 other) => this == other;
+
+    /// <summary>
+    /// Constructs a <see cref="DateTime"/> from this Instant which has a <see cref="DateTime.Kind" />
```

Comment:
Rather than `this Instant` (with that capitalisation), this should probably be one of:

- `this instant`
- `this instance`
- `this <c>Instant64</c>`

(likewise throughout, and same for e.g. `new Instant` below)